vllm/sequence.py

```diff
@@ -359,6 +362,7 @@ class SequenceGroupMetadata:
         sampling_params: The sampling parameters used to generate the outputs.
         block_tables: The block tables. (Seq id -> list of physical block
             numbers)
+        state: A dict for holding internal state tied to this sequence group.
```

Comment:
@Yard1 I'd intentionally kept this opaque to maintain separation of concerns since the torch generator is lower level and managed only by the model runner / sampler. If we define a dataclass here then the member type would be `torch.Generator` and as far as I can see `torch` is decoupled from the engine layer. In this case the model runner just needs access to somewhere tied to the lifecycle of the sequence group that it can stash the corresponding Generator.

But happy to change it if you're sure about this!

Reply:
It's mainly about ensuring we avoid errors caused by eg. typos in key names. Type hinting/checking is secondary

Reply:
@Yard1 have now changed it to use a dataclass